test/tools/ossfuzz/protoToYul.cpp

```diff
@@ -90,12 +90,33 @@ string ProtoConverter::visit(Literal const& _x)
 	}
 }
 
+void ProtoConverter::consolidateVarDeclsInFunctionDef()
+{
+	m_currentFuncVars.clear();
+	auto &scopes = m_funcVars.back();
+	for (auto &s: scopes)
+		m_currentFuncVars.insert(m_currentFuncVars.end(), s.begin(), s.end());
```

Comment:
Wouldn't it be better to work with pointers here? Also, do you really need to distinguish between function and global variables? Globals variables could just be variables in a function that is not explicitly defined.

Reply:
That's a good point. Perhaps, the data structure I should be using is a map of block pointer to variables declared in that block. I will need to then differentiate between function blocks and non-function blocks such that a referenced variable in a function body can come from any active scope within the function but not from outside the function.

Reply:
@chriseth I addressed the pointer comment, still thinking how to create a unified data structure for global/function variables.

Reply:
> Also, do you really need to distinguish between function and global variables?

Yes, in order to not reference the latter inside a function. Right now, it is implemented like so

https://github.com/ethereum/solidity/blob/e71cfb57baa9502aaaefc08f56c8e7ea63fa9d0b/test/tools/ossfuzz/protoToYul.cpp#L132-L142

Reply:
Why cannot `m_globalVars` be `m_funcVars[0]`?

Reply:
No strong reason against it. I felt separate vectors for global and function vars may make the code more readable than clubbing them together.